lib/client.js

```diff
@@ -48,10 +48,21 @@ async function get(config) {
     );
   }
 
+  const configTokenAuthMethod = config.tokenAuthMethod;
+  const issuerTokenAuthMethod = Array.isArray(issuer.token_endpoint_auth_methods_supported) ?
+    issuer.token_endpoint_auth_methods_supported : [];
+  if (configTokenAuthMethod && ! issuerTokenAuthMethod.includes(configTokenAuthMethod)) {
+    throw new Error(
+      `Token auth method "${configTokenAuthMethod}" is not supported by the issuer. ` +
+      `Supported token auth methods are "${issuerTokenAuthMethod.join('", "')}". `
+    );
+  }
+
```

Comment:
given that `token_endpoint_auth_methods_supported` is an optional attribute (which you handle) the throw is unexpected. With that, this whole diff can be removed.

Reply:
I'm not sure I get it. The attribute itself is optional, but it doesn't mean it shouldn't be checked against the issuer ? It works the exact same way as `response_modes` or `response_types` ?

Reply:
Consider the following

```js
const config = {
  tokenAuthMethod: 'client_secret_basic'
};
const issuer = {};

const configTokenAuthMethod = config.tokenAuthMethod;
  const issuerTokenAuthMethod = Array.isArray(issuer.token_endpoint_auth_methods_supported) ?
    issuer.token_endpoint_auth_methods_supported : [];
  if (configTokenAuthMethod && ! issuerTokenAuthMethod.includes(configTokenAuthMethod)) {
    throw new Error(
      `Token auth method "${configTokenAuthMethod}" is not supported by the issuer. ` +
      `Supported token auth methods are "${issuerTokenAuthMethod.join('", "')}". `
    );
  }
```

It should be possible to configure an explicit auth method without the issuer having the optional property exposed. At the moment, this code will throw and it shouldn't.

Given that the issuer discovery property is optional, this code is completely unnecessary.

Reply:
OK I get it. So is it better to completely remove this diff or check the attribute ONLY IF the issuer allows discovery ?

Reply:
completely remove